### PR TITLE
feat(meetings): add rejoin meeting capability

### PR DESCRIPTION
**Pull request by adamweeks**
image

PR consists of this section of the reconnect flow chart.

### CODE REVIEW

File: packages/node_modules/@webex/plugin-meetings/src/constants.js [hunk 1]
```diff
@@ -379,6 +379,7 @@ export const EVENT_TRIGGERS = {
   MEETING_RECONNECTION_STARTING: 'meeting:reconnectionStarting',
   MEETING_RECONNECTION_SUCCESS: 'meeting:reconnectionSuccess',
   MEETING_RECONNECTION_FAILURE: 'meeting:reconnectionFailure',
+  MEETING_RESHARE_NEEDED: 'meeting:reshareNeeded',
```
Comment (arun3528) [inline packages/node_modules/@webex/plugin-meetings/src/constants.js:382]: Should we just trigger share:stopped with reason as reconnect !?

File: packages/node_modules/@webex/plugin-meetings/src/config.js [hunk 2]
```diff
@@ -18,8 +18,10 @@ export default {
       detection: true,
       // Timeout duration to wait for ICE to reconnect if a disconnect is received.
       iceReconnectionTimeout: 10000,
+      // Amount of times attempting to rejoin a meeting during reconnect
+      maxRejoinAttempts: 3,
       retry: {
-        times: 2,
+        times: 5,
```
Comment (chrisadubois) [inline packages/node_modules/@webex/plugin-meetings/src/config.js:24]: Are we still using the backoff rate?

File: packages/node_modules/@webex/plugin-meetings/src/reconnection-manager/index.js [hunk 3]
```diff
@@ -173,7 +186,7 @@ export default class ReconnectionManager {
    * @public
    * @memberof ReconnectionManager
    */
-  reconnect({networkChange = false, networkDisconnect = false}) {
+  reconnect({networkChange = false, networkDisconnect = false} = {networkChange: false, networkDisconnect: false}) {
```
Comment (mwegman) [inline packages/node_modules/@webex/plugin-meetings/src/reconnection-manager/index.js:189]:
```suggestion
  reconnect({networkChange = false, networkDisconnect = false} = {}) {
```

Comment (mwegman) [reply packages/node_modules/@webex/plugin-meetings/src/reconnection-manager/index.js:189]: You don't need to define the object twice to get `reconnect()` to work.

File: packages/node_modules/@webex/plugin-meetings/src/constants.js (hunk 1 above)
Comment (adamweeks) [reply packages/node_modules/@webex/plugin-meetings/src/constants.js:382]: That works!

File: packages/node_modules/@webex/plugin-meetings/src/config.js (hunk 2 above)
Comment (adamweeks) [reply packages/node_modules/@webex/plugin-meetings/src/config.js:24]: Not sure why that was modified but I will remove all those configs not used.

File: packages/node_modules/@webex/plugin-meetings/src/reconnection-manager/index.js (hunk 3 above)
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/src/reconnection-manager/index.js]: I was wondering about the same

File: packages/node_modules/@webex/plugin-meetings/src/config.js [hunk 4]
```diff
@@ -18,13 +18,8 @@ export default {
...
       iceReconnectionTimeout: 10000,
-      retry: {
-        times: 2,
-        backOff: {
-          start: 1000,
-          rate: 2
-        }
-      }
+      // Amount of times attempting to rejoin a meeting during reconnect
+      maxRejoinAttempts: 3
```
Comment (InteractiveTimmy) [inline packages/node_modules/@webex/plugin-meetings/src/config.js:22]: Was this changed since back-off logic already exists elsewhere?

Comment (adamweeks) [reply packages/node_modules/@webex/plugin-meetings/src/config.js:22]: yes, this amount is specific to just the rejoin logic portion of the reconnect function.

**Pull request merged**